python/pyspark/mllib/linalg.py

```diff
@@ -821,6 +822,36 @@ def __reduce__(self):
             self.numRows, self.numCols, self.values.tostring(),
             int(self.isTransposed))
 
+    def __str__(self):
+        """
+        Pretty printing of a DenseMatrix
+
+        >>> dm = DenseMatrix(2, 2, range(4))
+        >>> print(dm)
+        0.0  2.0
+        1.0  3.0
+        >>> dm = DenseMatrix(2, 2, range(4), isTransposed=True)
+        >>> print(dm)
+        0.0  1.0
+        2.0  3.0
+        """
+        ds = []
+        for row in self.toArray():
+            ds.append("  ".join([str(col) for col in row]))
+        return "\n".join(ds)
```

Comment:
It is not trivial to make a pretty printing of a matrix. This implementation will fail on some thing like

~~~
1.234567 1.0 3.0
2.0 4.0 6.001
~~~

Can we use numpy's implementation directly? You can take a look at how `matrix.__str__` is implemented:

https://github.com/numpy/numpy/blob/master/numpy/matrixlib/defmatrix.py#L366